File: src/java/org/javaturk/dp/ch03/ocp/factory/solution/Manager.java
package org.javaturk.dp.ch03.ocp.factory.solution;

public class Manager extends Employee{
	protected String departmentManaged;
	
	public static final int MANAGEMENT_PAYMENT = 3000;
	
	public Manager(int no, String name, int year, String workingDepartment, String departmentManaged){
		super(no, name, year, workingDepartment);
		this.departmentManaged = departmentManaged;
	}
	
	public void work(){
		System.out.println("Manager is working!");
		manage();
	}
	
	public void manage(){
		System.out.println("Manager manages department: " + departmentManaged);
	}
	
	public double calculateSalary(){
		return super.calculateSalary() + MANAGEMENT_PAYMENT;
	}
	
	public String getDepartmentManaged() {
		return departmentManaged;
	}
	
	public void printInfo(){
		super.printInfo();
		System.out.println("Managing Department: " + departmentManaged);
	}

	@Override
	public String toString() {
		return "Manager [departmentManaged=" + departmentManaged + ", no=" + no + ", name=" + name + ", year=" + year
				+ ", department=" + department + ", salary=" + salary + "]";
	}
}

File: src/c#/Creational/FactoryMethod/FactoryMethod1/Manager.cs
using System;

namespace FactoryMethod1
{
    public class Manager : Employee
    {
		protected string departmentManaged { get;}

		public static readonly int MANAGEMENT_PAYMENT = 3000;

		public Manager(int no, string name, int year, string workingDepartment, string departmentManaged) : base(no, name, year, workingDepartment)
		{	
			this.departmentManaged = departmentManaged;
		}

		public override void Work()
		{
			Console.WriteLine("Manager is working!");
			Manage();
		}

		public virtual void Manage()
		{
			Console.WriteLine("Manager manages department: " + departmentManaged);
		}

		public override double CalculateSalary()
		{
			return base.CalculateSalary() + MANAGEMENT_PAYMENT;
		}
	}
}